lib/ruboclean.rb

```diff
@@ -12,8 +13,25 @@
 module Ruboclean
   def self.run_from_cli!(args)
     runner = Runner.new(args)
-    print "Using path '#{runner.path}' ... " if runner.verbose?
-    runner.run!
-    puts "done." if runner.verbose?
+    logger = Ruboclean::Logger.new(runner.verbose? ? :verbose : :none)
+
+    logger.verbose "Using path '#{runner.path}' ... "
+    changed = runner.run!
+    logger.verbose post_execution_message(changed, runner.verify?)
+
+    exit !changed if runner.verify?
```

Comment:
I don't use exit codes that often. Would it be bad if we returned `1` if the file changed and was actually written? I could live with "No backward compatibility" at this point.

Reply:
I think it is more preferred to exit with `0` when files are successfully changed. It's common to think `1` as failure. like shell, CI Jobs, Docker, etc. For example, when I create CI job that automatically runs `ruboclean`  and create PR to fix ordering, It displayed as failure when it exit with `1`.

In case of RuboCop, It returns 1 only when it executed with `verify mode`(I don't know about this name is right) and there are offenses.

dry, without offenses
```sh
❯ rubocop
Inspecting 28 files
............................

28 files inspected, no offenses detected
❯ echo $?
0
```

dry, with offenses
```sh
❯ rubocop
Inspecting 28 files
........C...................

Offenses:

lib/ruboclean/logger.rb:20:1: C: [Correctable] Layout/TrailingEmptyLines: 1 trailing blank lines detected.

28 files inspected, 1 offense detected, 1 offense autocorrectable
❯ echo $?
1
```

wet, without offenses
```sh
❯ rubocop -a
Inspecting 28 files
............................

28 files inspected, no offenses detected
❯ echo $?
0
```

wet, with offenses
```sh
❯ rubocop -a
Inspecting 28 files
........C...................

Offenses:

lib/ruboclean/logger.rb:20:1: C: [Corrected] Layout/TrailingEmptyLines: 1 trailing blank lines detected.

28 files inspected, 1 offense detected, 1 offense corrected
❯ echo $?
0
```

Reply:
I think it is more preferred to exit with `0` when files are successfully changed. It's common to think `1` as failure. like shell, CI Jobs, Docker, etc. For example, when I create CI job that automatically runs ruboclean and create PR to fix ordering, It displayed as failure when it exit with `1`.

In case of RuboCop, It returns `1` only when it executed with verify mode(I don't know about this name is right) and there are offenses.

dry, without offenses
```sh
❯ rubocop
Inspecting 28 files
............................

28 files inspected, no offenses detected
❯ echo $?
0
```

dry, with offenses
```sh
❯ rubocop
Inspecting 28 files
........C...................

Offenses:

lib/ruboclean/logger.rb:20:1: C: [Correctable] Layout/TrailingEmptyLines: 1 trailing blank lines detected.

28 files inspected, 1 offense detected, 1 offense autocorrectable
❯ echo $?
1
```

wet, without offenses
```sh
❯ rubocop -a
Inspecting 28 files
............................

28 files inspected, no offenses detected
❯ echo $?
0
```

wet, with offenses
```sh
❯ rubocop -a
Inspecting 28 files
........C...................

Offenses:

lib/ruboclean/logger.rb:20:1: C: [Corrected] Layout/TrailingEmptyLines: 1 trailing blank lines detected.

28 files inspected, 1 offense detected, 1 offense corrected
❯ echo $?
0
```

Reply:
Wow, thank you for your explanations. I think I now understand better how exit codes could/should be used.